### ISSUE TITLE
Handle errors in Google Cloud fixity checking

### ISSUE THREAD

**Issue by escowles** [pulibrary/figgy#3285]
Currently, if there is a fixity error when downloading and checking the checksum of a file, the client errors and no error is logged.  This can only happen in a couple of narrow circumstances:

  1. Transitory errors in downloading a file (the vast majority of cases where this would happen)
  1. Bitrot in Google Cloud storage
  1. Mismatch between the Figgy and Google checksums for a file

We should catch and retry errors in order to confirm errors (and distinguish them from transitory downloading problems), so we can log them appropriately.

**Comment by escowles** [comment]
We should retry 5 times — probably going to take care of intermittent issues, and we can't retry enough times to avoid major network issues.  If we can delay, that might help resolve network issues.

**Comment by hackmastera** [comment]
The 3rd case is handled here, but is anything in figgy checking this pubsub later?

https://github.com/pulibrary/figgy/blob/4178105737a948c7f02722af360d28f9df76a3c3/cloud_fixity/index.js#L40

**Comment by hackmastera** [comment]
oh yes I see it in `app/services/cloud_fixity`

**Comment by tpendragon** [comment]
We had talked about delaying the retries, but @hackmastera and I came to the conclusion that we shouldn't try to do that right now for the following reasons:

1. The network problems would have to be purely internal to Google to cause a problem, which is unlikely.
2. Cloud Functions auto-acknowledge messages on receive, so to implement we'd have to add something to the function that checks if it's been enough time and re-queues if it hasn't, which would mean we'd have to pay for that failed worker's time basically non-stop until it was ready to retry again, when an immediate retry may fix it (or it may even be a legitimate error we should look into.)
3. If there WERE some big failure of Google's Storage infrastructure, we'd be able to just find all those events that happened in that window and either retry fixity check on those (resulting in their removal from our fixity dashboard), or just delete them.
